src/main/java/org/opensearch/neuralsearch/processor/NormalizationProcessorWorkflow.java

```diff
@@ -202,20 +205,46 @@ private void updateOriginalQueryResults(final CombineScoresDto combineScoresDTO,
         final List<CompoundTopDocs> queryTopDocs = getCompoundTopDocs(combineScoresDTO, querySearchResults);
         final Sort sort = combineScoresDTO.getSort();
         int totalScoreDocsCount = 0;
-        for (int index = 0; index < querySearchResults.size(); index++) {
-            QuerySearchResult querySearchResult = querySearchResults.get(index);
-            CompoundTopDocs updatedTopDocs = queryTopDocs.get(index);
-            totalScoreDocsCount += updatedTopDocs.getScoreDocs().size();
-            TopDocsAndMaxScore updatedTopDocsAndMaxScore = new TopDocsAndMaxScore(
-                buildTopDocs(updatedTopDocs, sort),
-                maxScoreForShard(updatedTopDocs, sort != null)
+
+        // Get index of first non-empty CompoundTopDocs to check if collapse is enabled
+        boolean isCollapseEnabled = false;
+        int indexOfFirstNonEmpty = 0;
```

Comment:
We should keep this as `-1` if not found, otherwise it will always pick up index 0

Reply:
Wouldn't it be the equal to the length of queryTopDocs if nothing is found?

Reply:
Changed it per suggestions below